bottlecap/src/telemetry/listener.rs

```diff
@@ -161,15 +201,10 @@ impl TelemetryListener {
     }
 
     fn handle_stream(
-        stream: &mut impl Read,
-        mut buf: [u8; 262_144],
+        stream: &TcpStream,
```

Comment:
ah yeah I made this change in my branch too, a bit easier to handle than receiving a generic which implements Read.